reckless_lib/src/utils.rs

```diff
@@ -23,19 +25,65 @@ pub fn create_dir_in_home(relative_path: &str) {
     };
 }
 
-pub fn get_dir_path_from_url(url: &str) -> String {
-    let path = env::home_dir()
+pub fn get_reckless_dir(dir: &str) -> String {
+    let mut path = env::home_dir()
         .unwrap()
         .into_os_string()
         .into_string()
         .unwrap();
+    path = format!("{}/.reckless/{}", path, dir);
+    path
+}
+
+pub fn get_dir_path_from_url(url: &str) -> String {
     return format!(
-        "{}/.reckless{}",
-        path,
+        "{}/{}",
+        get_reckless_dir("repositories"),
         &url.split(".com").last().unwrap().trim().to_string()
     );
 }
 
+fn slice_from_end(string: &str, size: usize) -> Option<&str> {
+    string
+        .char_indices()
+        .rev()
+        .nth(size)
+        .map(|(i, _)| &string[i..])
+}
+
+pub fn get_repo_name_from_url(url: &str) -> String {
+    let mut repo_name = url.split("/").last().unwrap().to_string();
+    if slice_from_end(url, 3).unwrap().to_string() == ".git" {
+        repo_name = repo_name.strip_suffix(".git").unwrap().to_string();
+    };
+    repo_name
+}
+
+pub fn index_reckless_dir(url: &str) -> Vec<Plugin> {
+    let mut list_plugins = Vec::<Plugin>::new();
+    let repo_path = get_dir_path_from_url(url);
+    let repo_name = get_repo_name_from_url(url);
+    let pattern = format!("{}/**/*", &repo_path);
+    for entry in glob(&pattern).unwrap() {
+        match entry {
+            Ok(path) => {
+                let file_name = path.file_name().unwrap().to_str().unwrap();
+                match file_name {
+                    "requirements.txt" => debug!("PYTHON"),
+                    "package.json" => debug!("JS"),
+                    "pubspec.yaml" => debug!("DART"),
+                    "cargo.toml" => debug!("RUST"),
+                    "tsconfig.json" => debug!("TS"),
+                    _ => print!(""),
+                }
+                list_plugins.push(Plugin::new(&repo_name, &path.to_str().unwrap().to_string()));
```

Comment:
2# Also Here I wish to only add to the `Vec` when we detect a `requirements.txt` or `package.json`, etc. But would that mean I will have to do this individually for each one? Because with the current approach there will be a lot of duplicates and logically it would be incorrect.